RestApiHomeTask/Controllers/ItemController.cs

```diff
@@ -0,0 +1,120 @@
+using Microsoft.AspNetCore.Mvc;
+using Npgsql;
+using RestApiHomeTask;
+
+namespace RestApiHomeTask.Controllers;
+
+[ApiController]
+[Route("api/[controller]/[action]")]
+[Consumes("application/json")]
+public class ItemController : ControllerBase
+{
+    NpgsqlConnection conn = new NpgsqlConnection("Host=localhost; Database=HomeTaskDB; Username=postgres; Password=adminS;");
+
+    [HttpGet(Name = "GetItemList")]
+    public IEnumerable<ItemVMModel> Get()
+    {
+
+        conn.Open();
+        // Passing PostGre SQL Function Name
+        NpgsqlCommand command = new NpgsqlCommand("Select * From items", conn);
+        // Execute the query and obtain a result set
+        NpgsqlDataReader reader = command.ExecuteReader();
+
+        List<ItemVMModel> listOfItems = new List<ItemVMModel>();
+        
+        while (reader.Read())
+        {
+          ItemVMModel temp =  new ItemVMModel
+            {
+                Id = reader["id"].ToString(),
+                Name = reader["name"].ToString(),
+                Categoryid = reader["categoryId"].ToString(),
+                Details = reader["details"].ToString()
+            }; // Remember Type Casting is required here it has to be according to database column data type
+            listOfItems.Add(temp);
+        }
+        reader.Close();
+        
+        command.Dispose();
+        conn.Close();
+        return listOfItems.ToArray();
+    }
+    [HttpPost(Name = "ListItemsFilters")]
+    public IEnumerable<ItemVMModel> ListItemsFilters(ItemFilter filter)
+    {
+         conn.Open();
+         var whereClauseCategoryId = filter.Categoryid != null ?  $"WHERE categoryid = '{filter.Categoryid}'" : " ";
+         var orderBy = filter.PageNumber != null && filter.PageSize != null ?  ", ROW_NUMBER() OVER(ORDER BY (SELECT NULL)) AS RN"  :  ""; 
+         var addedOrderBy = filter.PageNumber != null && filter.PageSize != null ? "ORDER BY RN" : "";
+         var pageSettings = filter.PageNumber != null && filter.PageSize != null ? $"OFFSET {filter.PageNumber} ROWS FETCH NEXT {filter.PageSize} ROW ONLY" : " ";
+
+        // Passing PostGre SQL Function Name
+        NpgsqlCommand command = new NpgsqlCommand($"Select * {orderBy} From items {whereClauseCategoryId} {addedOrderBy} {pageSettings} ", conn);
+        // Execute the query and obtain a result set
+        NpgsqlDataReader reader = command.ExecuteReader();
```

Comment:
reader is disposable so you can use using statement for it, and it will be safer in case of errors